urfave/build.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2019, the Drone Plugins project authors.
+// Please see the AUTHORS file for details. All rights reserved.
+// Use of this source code is governed by an Apache 2.0 license that can be
+// found in the LICENSE file.
+
+package urfave
+
+import (
+	"time"
+
+	"github.com/drone-plugins/drone-plugin-lib/drone"
+	"github.com/urfave/cli/v2"
+)
+
+// buildFlags has the cli.Flags for the drone.Build.
+func buildFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringFlag{
+			Name:    "build.branch",
+			Usage:   "build branch",
+			EnvVars: []string{"DRONE_BRANCH"},
+		},
+		&cli.IntFlag{
+			Name:    "build.pull-request",
+			Usage:   "build pull request",
+			EnvVars: []string{"DRONE_PULL_REQUEST"},
+		},
+		&cli.StringFlag{
+			Name:    "build.tag",
+			Usage:   "build tag",
+			EnvVars: []string{"DRONE_TAG"},
+		},
+		&cli.StringFlag{
+			Name:    "build.source-branch",
+			Usage:   "build source branch",
+			EnvVars: []string{"DRONE_SOURCE_BRANCH"},
+		},
+		&cli.StringFlag{
+			Name:    "build.target-branch",
+			Usage:   "build target branch",
+			EnvVars: []string{"DRONE_TARGET_BRANCH"},
+		},
+		&cli.IntFlag{
+			Name:    "build.number",
+			Usage:   "build number",
+			EnvVars: []string{"DRONE_BUILD_NUMBER"},
+		},
+		&cli.IntFlag{
+			Name:    "build.parent",
+			Usage:   "build parent",
+			EnvVars: []string{"DRONE_BUILD_PARENT"},
+		},
+		&cli.StringFlag{
+			Name:    "build.event",
+			Usage:   "build event",
+			EnvVars: []string{"DRONE_BUILD_EVENT"},
+		},
+		&cli.StringFlag{
+			Name:    "build.action",
+			Usage:   "build action",
+			EnvVars: []string{"DRONE_BUILD_ACTION"},
+		},
+		&cli.StringFlag{
+			Name:    "build.status",
+			Usage:   "build status",
+			EnvVars: []string{"DRONE_BUILD_STATUS"},
+		},
+		&cli.StringFlag{
+			Name:    "build.created",
+			Usage:   "build created",
+			EnvVars: []string{"DRONE_BUILD_CREATED"},
+		},
+		&cli.StringFlag{
+			Name:    "build.started",
+			Usage:   "build started",
+			EnvVars: []string{"DRONE_BUILD_STARTED"},
+		},
+		&cli.StringFlag{
+			Name:    "build.finished",
+			Usage:   "build finished",
+			EnvVars: []string{"DRONE_BUILD_FINISHED"},
+		},
+		&cli.StringFlag{
+			Name:    "build.deploy-to",
+			Usage:   "build deploy to",
+			EnvVars: []string{"DRONE_DEPLOY_TO"},
+		},
+		&cli.StringSliceFlag{
```

Comment:
Would be nice if https://github.com/urfave/cli/issues/603 was fixed so then we can just have this just place everything there so we don't need `buildFromContext`

Reply:
Yes, it would be nice, but the issue is already quite long open and I don't know when this will be finally resolved.